src/actions/indexing/indexable-post-indexation-action.php

```diff
@@ -55,15 +55,17 @@ public function __construct( Post_Type_Helper $post_type_helper, Indexable_Repos
 	/**
 	 * Returns the total number of unindexed posts.
 	 *
+	 * @param int $limit Limit the number of unindexed posts that are counted.
+	 *
 	 * @return int|false The total number of unindexed posts. False if the query fails.
 	 */
-	public function get_total_unindexed() {
+	public function get_total_unindexed( $limit = false ) {
 		$transient = \get_transient( static::TRANSIENT_CACHE_KEY );
 		if ( $transient !== false ) {
 			return (int) $transient;
 		}
 
-		$query = $this->get_query( true );
+		$query = $this->get_count_query( $limit );
 
 		$result = $this->wpdb->get_var( $query );
```

Comment:
If we make `get_count_query() `to return rows instead of count() (as per my previous comment), then here we should count those rows (all that to cover the "`select ID's (LIMIT shutdown_limit) order by ID ` instead of counting the entire indexables in the "count" queries" item).

So, here we should
```suggestion
		$result = count( $this->wpdb->get_results( $query ) );
```